openmc/plots.py

```diff
@@ -0,0 +1,850 @@
+            msg = 'Unable to set the meshlines with ' \
+                  'type "{0}"'.format(meshlines['type'])
+            raise ValueError(msg)
+
+        if 'id' in meshlines:
+            cv.check_type('plot meshlines id', meshlines['id'], Integral)
+            cv.check_greater_than('plot meshlines id', meshlines['id'], 0,
+                                  equality=True)
+
+        if 'linewidth' in meshlines:
+            cv.check_type('plot mesh linewidth', meshlines['linewidth'], Integral)
+            cv.check_greater_than('plot mesh linewidth', meshlines['linewidth'],
+                                  0, equality=True)
+
+        if 'color' in meshlines:
+            self._check_color('plot meshlines color', meshlines['color'])
+
+        self._meshlines = meshlines
+
+    @staticmethod
+    def _check_color(err_string, color):
+        cv.check_type(err_string, color, Iterable)
+        if isinstance(color, str):
+            if color.lower() not in _SVG_COLORS:
+                raise ValueError("'{}' is not a valid color.".format(color))
+        else:
+            cv.check_length(err_string, color, 3)
+            for rgb in color:
+                cv.check_type(err_string, rgb, Real)
+                cv.check_greater_than('RGB component', rgb, 0, True)
+                cv.check_less_than('RGB component', rgb, 256)
+
+    def __repr__(self):
+        string = 'Plot\n'
+        string += '{: <16}=\t{}\n'.format('\tID', self._id)
+        string += '{: <16}=\t{}\n'.format('\tName', self._name)
+        string += '{: <16}=\t{}\n'.format('\tFilename', self._filename)
+        string += '{: <16}=\t{}\n'.format('\tType', self._type)
+        string += '{: <16}=\t{}\n'.format('\tBasis', self._basis)
+        string += '{: <16}=\t{}\n'.format('\tWidth', self._width)
+        string += '{: <16}=\t{}\n'.format('\tOrigin', self._origin)
+        string += '{: <16}=\t{}\n'.format('\tPixels', self._pixels)
+        string += '{: <16}=\t{}\n'.format('\tColor by', self._color_by)
+        string += '{: <16}=\t{}\n'.format('\tBackground', self._background)
+        string += '{: <16}=\t{}\n'.format('\tMask components',
+                                            self._mask_components)
+        string += '{: <16}=\t{}\n'.format('\tMask background',
+                                            self._mask_background)
+        string += '{: <16}=\t{}\n'.format('\Overlap Color',
+                                            self._overlap_color)
+        string += '{: <16}=\t{}\n'.format('\tColors', self._colors)
+        string += '{: <16}=\t{}\n'.format('\tLevel', self._level)
+        string += '{: <16}=\t{}\n'.format('\tMeshlines', self._meshlines)
+        return string
+
+    @classmethod
+    def from_geometry(cls, geometry, basis='xy', slice_coord=0.):
+        """Return plot that encompasses a geometry.
+
+        Parameters
+        ----------
+        geometry : openmc.Geometry
+            The geometry the base the plot off of
```

Comment:
Spelling: I believe you may have meant "The geometry **to** base the plot off of"